yudiencore/core.go

```diff
@@ -310,6 +327,25 @@ func UdnLog(udn_schema map[string]interface{}, format string, args ...interface{
 	}
 }
 
+func UdnLogLevel(udn_schema map[string]interface{}, log_level int, format string, args ...interface{}) {
+	// Function works the same as UdnLog/UdnError but allows level logging
+	//TODO(z): Migrate UdnDebug functionality here later
+	//TODO(z): Combine all log functions to put under UdnLogLevel
+	if log_level <= Debug_Udn_Log_Level {
+		output := fmt.Sprintf(format, args...)
+
+		fmt.Print(output)
```

Comment:
with `fmt.Printf`, it should be possible to encapsulate the Sprintf and Print together

Reply:
Done